src/decrypter.js

```diff
@@ -158,14 +84,160 @@ export class Decrypter {
+ *
+ * @return {WebCrypto Object}
+ */
+const getWebCrypto = function() {
+  let _crypto = window ? window.crypto || window.msCrypto : crypto;
+
+  if (_crypto.webkitSubtle) {
+    _crypto.subtle = _crypto.webkitSubtle;
+  }
+
+  return _crypto.subtle ? _crypto : null;
 };
+
+/**
+ * Decrypt bytes using AES-128 with CBC and PKCS#7 padding.
+ *
+ * @param {Uint8Array} encrypted the encrypted bytes
+ * @param {Uint8Array} key the bytes of the decryption key
+ * @param {Uint32Array} iv the initialization vector (IV) to
+ * use for the first round of CBC.
+ * @param {Function} done callback that takes a Uint8Array
+ *
+ * @see http://en.wikipedia.org/wiki/Advanced_Encryption_Standard
+ * @see http://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Cipher_Block_Chaining_.28CBC.29
+ * @see https://tools.ietf.org/html/rfc2315
+ */
+const decryptNative = function(encrypted, key, iv, done) {
+  let crypto = getWebCrypto();
+  let algorithm = {name: 'AES-CBC', iv};
+  let extractable = true;
+  let usages = ['decrypt'];
+
+  let keyPromise = crypto.subtle.importKey('raw', key, algorithm, extractable, usages);
+
+  return keyPromise.then(function(importedKey) {
+    return crypto.subtle.decrypt(algorithm, importedKey, encrypted);
+  }).catch(function(rejection) {
+    return done(null, new Uint8Array());
+  }).then(function(plaintextArrayBuffer) {
+    return done(null, new Uint8Array(plaintextArrayBuffer));
+  });
+};
+
+/**
+ * Decrypt bytes using AES-128 with CBC and PKCS#7 padding.
+ *
+ * @param {Uint8Array} encrypted the encrypted bytes
+ * @param {Uint8Array} key the bytes of the decryption key
+ * @param {Uint32Array} iv the initialization vector (IV) to
+ * use for the first round of CBC.
+ * @return {Promise} that resolves with a Uint8Array the decrypted bytes
+ *
+ * @see http://en.wikipedia.org/wiki/Advanced_Encryption_Standard
+ * @see http://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Cipher_Block_Chaining_.28CBC.29
+ * @see https://tools.ietf.org/html/rfc2315
+ */
+const decryptNonNativeWithDecrypter = function(encrypted, key, iv, done) {
+  return new Decrypter(encrypted, key, iv, done);
+};
+
+/**
+ * Decrypt bytes using AES-128 with CBC and PKCS#7 padding.
+ * Chooses webcrypto or JS implementation where available.
+ *
+ * @param {Uint8Array} encrypted: the encrypted bytes
+ * @param {Uint8Array} key: the bytes of the decryption key
+ * @param {Uint32Array} iv: the initialization vector (IV) to
+ * use for the first round of CBC.
+ *
+ * @see http://en.wikipedia.org/wiki/Advanced_Encryption_Standard
+ * @see http://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Cipher_Block_Chaining_.28CBC.29
+ * @see https://tools.ietf.org/html/rfc2315
+ */
+export const decrypt = function(encrypted, key, iv, done) {
+  let decryptionMethod = getWebCrypto() ? decryptNative : decryptNonNativeWithDecrypter;
+
+  return decryptionMethod(encrypted, key, iv, done);
+};
+
+export default decrypt;
```

Comment:
The Decrypter was an object because there is state that has to be regenerated per key-iv combo. Our usage of this project is very performance sensitive so I think the old API is justified.